engine/src/main/java/org/terasology/network/BroadCastServer.java

```diff
@@ -0,0 +1,131 @@
+    private static final String DISCOVERY_REQUEST = "DISCOVERY_REQUEST";
+    private static final String DISCOVERY_RESPONSE = "DISCOVERY_RESPONSE";
+
+    public boolean isBroadCastTurnedOn() {
+        return turnOnBroadcast;
+    }
+
+    public void startBroadcast() {
+        BroadCastServer.turnOnBroadcast = true;
+        try {
+            sendBroadCast();
+            Executors.newSingleThreadExecutor().execute(new Runnable() {
+                @Override
+                public void run() {
+                    while (true) {
+                        try {
+                            while (turnOnBroadcast) {
+                                listenToBroadCast();
+                            }
+                        } catch (Exception e) {
+                            logger.error("Broadcasting has been interrupted" + e.getMessage());
+                        }
+                    }
+                }
+            });
+        } catch (Exception e) {
+        logger.info("Broadcasting has been interrupted " + e.getMessage());
+        }
+    }
+
+    public void stopBroadCast() {
+        BroadCastServer.turnOnBroadcast = false;
+        logger.info("Shutting down BroadCasting");
+        service.shutdown();
+    }
+
+    private void listenToBroadCast() throws Exception {
+        byte[] buffer = DISCOVERY_REQUEST.getBytes(StandardCharsets.UTF_8);
+        DatagramPacket packet = new DatagramPacket(buffer, buffer.length);
+        getReceiveSocket().receive(packet);
+        logger.info("Discovery package received! -> " + packet.getAddress() + ":" + packet.getPort());
+
+        //Validate the data sent.
+        String data = new String(packet.getData()).trim();
+        if (data.equals(DISCOVERY_REQUEST)) { // validate command
+            // Send response
+            byte[] response = new byte[DISCOVERY_RESPONSE.length()];
+            DatagramPacket responsePacket = new DatagramPacket(response, response.length, packet.getAddress(), packet.getPort());
+            getReceiveSocket().send(responsePacket);
+            logger.info("Response sent to: " + packet.getAddress() + ":" + packet.getPort());
+        } else {
+            logger.info("Error, not valid package!" + packet.getAddress() + ":" + packet.getPort());
+        }
+    }
+
+    private DatagramSocket getReceiveSocket()  {
+        try {
+            if (receiveSocket == null) {
+                receiveSocket = new DatagramSocket(8002,
+                    InetAddress.getByName("0.0.0.0")); // 0.0.0.0 for listen to all ips
+                receiveSocket.setBroadcast(true);
+            }
+        } catch (Exception e) {
+            logger.error("Broadcast Exception Encountered" + e.getMessage());
+        }
+        return receiveSocket;
+    }
+
+    private DatagramSocket getSendSocket() {
+        try {
+            if (sendSocket == null) {
+                sendSocket = new DatagramSocket(8001, InetAddress.getLocalHost());
```

Comment:
Seems, it is not working in LAN. Try test this PR on several computers.

Reply:
> Better separate area with something like 'LAN' title. Or something like separate Tab (if it is possible).
> When I saw your PR I thought about Counter-Strike 1.6's 'LAN Games' :)

Yeah will work on this :)

Reply:
I tried this on wire shark and it does capture the packets but I have no access to more computers, could someone else try it

Reply:
If you using linux or msys, you can try to use 'nc' tool.
You can try to use docker or virtual machine, don't forget setup network.

Reply:
If you using linux or msys, you can try to use 'nc' tool.
You can try to use docker or virtual machine, don't forget setup network.

Reply:
Yeah tried it on two computers it does show up :) Can anyone confirm